src/parquet/column/reader.h

```diff
@@ -184,7 +188,7 @@ inline int64_t TypedColumnReader<DType>::ReadBatch(int batch_size, int16_t* def_
   }
 
   // Not present for non-repeated fields
-  if (descr_->max_repetition_level() > 0) {
+  if (descr_->max_repetition_level() > 0 && rep_levels) {
     num_rep_levels = ReadRepetitionLevels(batch_size, rep_levels);
     if (num_def_levels != num_rep_levels) {
       throw ParquetException("Number of decoded rep / def levels did not match");
```

Comment:
My problem is is that I couldn't get Spark to write columns with no definition levels. Instead I get columns with `max_definition_level=1`, and `def_devels` set to all ones. And the JSON schema says `"nullable":true`, which is the opposite of what I asked.

The fix should really be made against Spark.  But it was trivial to apply a workaround to parquet-cpp.